### ISSUE TITLE
Descriptions of config codes

### ISSUE THREAD

**Issue by rustyx** [thebookisclosed/ViVe#12]
Is there any way to get any information on what the various configs do?

There are more and more of them, e.g.:

```
> .\ViveTool.exe queryconfig all

[0x011738FA / 18299130]
Group: 4
EnabledState: Enabled (2)
EnabledStateOptions: 1
Variant: 0
VariantPayloadKind: 0
VariantPayload: 0

[0x012181C3 / 18973123]
Group: 4
EnabledState: Enabled (2)
EnabledStateOptions: 1
Variant: 1
VariantPayloadKind: 1
VariantPayload: a900000

[0x012BAA03 / 19638787]
Group: 4
EnabledState: Enabled (2)
EnabledStateOptions: 1
Variant: 0
VariantPayloadKind: 0
VariantPayload: 0

[0x0140810C / 21004556]
Group: 4
EnabledState: Enabled (2)
EnabledStateOptions: 1
Variant: 1
VariantPayloadKind: 1
VariantPayload: 3c3cb4

[0x014F38C9 / 21969097]
Group: 4
EnabledState: Enabled (2)
EnabledStateOptions: 1
Variant: 1
VariantPayloadKind: 1
VariantPayload: 1fd4c33

[0x015FD432 / 23057458]
Group: 4
EnabledState: Enabled (2)
EnabledStateOptions: 1
Variant: 1
VariantPayloadKind: 1
VariantPayload: 600000f

[0x01670E38 / 23531064]
Group: 4
EnabledState: Enabled (2)
EnabledStateOptions: 1
Variant: 2
VariantPayloadKind: 0
VariantPayload: 0
. . .
```

I'm asking because I tried to enable config 18299130 to hide the settings header, but it had no effect, so I presume the code has changed in 21H1.

If not, does anyone know at least which files are responsible for handling the config codes so I can try to peek inside?

**Comment by hpwamr** [comment]
> I'm asking because I tried to enable config 18299130 to hide the settings header, but it had no effect, so I presume the code has changed in 21H1.

Yesterday, removed without any problem the "Settings Header Banner" on 2x Win10 21H1. 🤔 
```
Edition	Windows 10 Pro
Version	21H1
Installed on	‎29/‎09/‎2020
OS build	19043.1023
Experience	Windows Feature Experience Pack 120.2212.2020.0
```

**Comment by nochlab1** [comment]
You have to open powershell as admin

**Comment by rustyx** [comment]
Ok setting 18299130 to 1 indeed works.

But the question is more general - how to know which code does what, short of disassembling Windows DLLs and tracing each one.

I found ~600 Windows DLLs that call RtlQueryFeatureConfiguration. So I guess it's possible to at least make a mapping of codes to DLLs, to at least get an idea about the area in which a code is used.

**Comment by Beerfloat** [comment]
> > I'm asking because I tried to enable config 18299130 to hide the settings header, but it had no effect, so I presume the code has changed in 21H1.
> 
> Yesterday, removed without any problem the "Settings Header Banner" on 2x Win10 21H1. 👍
> 
> ```
> Edition	Windows 10 Pro
> Version	21H1
> Installed on	‎29/‎09/‎2020
> OS build	19043.1023
> Experience	Windows Feature Experience Pack 120.2212.2020.0
> ```

Sadly removing the settings header banner doesn't seem to work anymore on 19043.1151 (2021-07 Cumulative Update Preview).

Microsoft seems to think 'Rewards - Start Earning' is something an adult wants to see in his operating system settings page.

**Comment by thebookisclosed** [comment]
The problem you are experiencing will now be tracked by issue #15 

I'm keeping this thread open due to an upcoming feature which lets ViVeTool use a Name<->ID dictionary

**Comment by jnpl95** [comment]
> The problem you are experiencing will now be tracked by issue #15
> 
> I'm keeping this thread open due to an upcoming feature which lets ViVeTool use a Name<->ID dictionary

Are you reverse engineering every name for each id? Or does microsoft list this info somewhere online?

**Comment by ericgl** [comment]
Yes, a description for each feature would be very welcome.

**Comment by the-j0k3r** [comment]
like @ericgl said, this would be helpful and remains unanswered. but I guess only someone who is familiar with the API's used would know, 

small rant;

I doubt MicroControlFreaksSoft (Microsoft) would make any documentation on this and its clear they will do anything possible to prevent people from Making the products they buy from MicroControlFreaksSoft to make the product into a usable,bloat free, nonsense free, need to come goto 3rd parties who, thank the coding gods, are inventive enough to come up with tools like this.

And then MicroControlFreaksSoft will make such solutions unusable just because pissing off their non corporate customers and using them and their data for making more billions is more important than actual user satisfaction.

Corporate greed and clueless product managers FTW.

Sorry for rant, I despise Corporations and MS in general more so for what they continue doing.

**Comment by thebookisclosed** [comment]
This feature is present in the newly released v0.3.0.
